### PR TITLE
Stop folding options and flags to camel case

### LINKED ISSUE #367: Hyphenated flags are not supported correctly
If I want to define a hyphenated flag (eg: 'no-start'), I need to define the `.spec.ts` as:

```ts
export interface Input {
   flags: {
      'no-start'?: boolean;
   };
}

const spec: CommandSpec<Input> = {
   description: 'Do something',
   flags: {
      'no-start': {
         description: 'Skip the start step',
      },
   },
};

export default spec;
```

But, when I then use this, and output the received data object, I see:

```json
flags: {
   noStart: true
}
```

The `no-start` has been transformed into `noStart`.

Seems good, but, if I define the interface (above) instead with:

```ts
   noStart?: boolean,
```

I receive an error message, complaining that `noStart` does not match up with anything in the spec (it's expecting `no-start`).

```
@sparksuite/cli:   Object literal may only specify known properties, and ''no-start'' does not exist in type
```

The issue seems to stem from the items in the list of descriptions *not* being translated from camel case into kebab case before being displayed.

### PR DESCRIPTION
**Pull request by pbelbin**
Closes #367 

Conversion of flags and options to camelCase has an impact on the ability to perform type checking, so it's been decided to not do the conversion any more.

### CODE REVIEW

File: src/utils/construct-input-object.test.ts [hunk 1]
```diff
@@ -37,6 +37,32 @@ describe('#constructInputObject()', () => {
...
+				'delivery-zip-code': undefined,
+				test: undefined,
+			},
+			flags: {
+				quiet: false,
+			},
+			passThroughArgs: ['--pass-through-flag', 'pass-through-option=value', 'pass-through-data'],
+		});
+	});
+
```
Comment (WesCossick) [inline src/utils/construct-input-object.test.ts:65]: Should this test be added to this PR? Unless I'm missing something, it seems unrelated.

Comment (pbelbin) [reply src/utils/construct-input-object.test.ts:65]: Without this test being added, the changes appear to expose some code to a lack of test coverage, so I added this to help keep the coverage score up.

**Review by WesCossick — APPROVED** [review]
